controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -283,6 +298,81 @@ func (ew *EndpointsWatcher) deleteEndpoints(obj interface{}) {
 	}
 }
 
+func (ew *EndpointsWatcher) addEndpointSlice(obj interface{}) {
+	es, ok := obj.(*discovery.EndpointSlice)
+	if !ok {
+		ew.log.Errorf("error processing EndpointSlice resource, got %T expected *discovery.EndpointSlice", obj)
+		return
+	}
+
+	if es.Namespace == kubeSystem {
+		return
+	}
+	id, err := getEndpointSliceServiceID(es)
+	if err != nil {
+		ew.log.Errorf("Could not fetch service name for EndpointSlice [%s/%s]:%v", es.Namespace, es.Name, err)
+		return
+	}
+	sp := ew.getOrNewServicePublisher(id)
+	sp.addEndpointSlice(es)
+}
+
+func (ew *EndpointsWatcher) updateEndpointSlice(oldObj interface{}, newObj interface{}) {
+	oldSlice, ok := oldObj.(*discovery.EndpointSlice)
+	if !ok {
+		ew.log.Errorf("error processing EndpointSlice resource, got %#v expected *discovery.EndpointSlice", oldObj)
+		return
+	}
+	newSlice, ok := newObj.(*discovery.EndpointSlice)
+	if !ok {
+		ew.log.Errorf("error processing EndpointSlice resource, got %#v expected *discovery.EndpointSlice", oldObj)
+		return
+	}
+
+	if newSlice.Namespace == kubeSystem {
+		return
+	}
+
+	id, err := getEndpointSliceServiceID(newSlice)
+	if err != nil {
+		ew.log.Errorf("Could not fetch service name for EndpointSlice [%s/%s]:%v", newSlice.Namespace, newSlice.Name, err)
```

Comment:
Not sure there is a need to chain errors like that. Why not simply format the error you return from `getEndpointSliceServiceID` Seems like if you include the slice name and namespace it would give us all the info we need.

Reply:
This is a great suggestion! I have made the changes. 

* In addition to moving the `ns/name` identifiers in `getEndpointSliceServiceID`, I have also modified `isValidSlice` to return a less verbose error, so it can be chained in a better way in `getEndpointSliceServiceID`
* When we process the slice, the error message in my opinion should say that the slice could not be processed, it might be more helpful in the debugging process to know that's where the error was thrown (speaking from experience, at least).

Thank you, let me know if you see them done/structured in a different way and will make the changes :)